controllers/src/main/scala/cromwell/pipeline/controller/ProjectFileController.scala

```diff
@@ -28,24 +31,59 @@ class ProjectFileController(wdlService: ProjectFileService)(
       },
       path("files") {
-        post {
-          entity(as[ProjectUpdateFileRequest]) {
-            request =>
-              onComplete(for {
-                validateResponse <- wdlService.validateFile(FileContent(request.projectFile.content))
-                uploadResponse <- wdlService.uploadFile(request.project, request.projectFile, request.version)
-              } yield {
-                (validateResponse, uploadResponse) match {
-                  case (Right(_), Right(response)) => StatusCodes.OK.intValue -> response
-                  case (Left(_), Right(response))  => StatusCodes.Created.intValue -> response
-                  case (_, Left(response))         => StatusCodes.UnprocessableEntity.intValue -> response.message
+        concat(
+          post {
+            entity(as[ProjectUpdateFileRequest]) {
+              request =>
+                onComplete(for {
+                  validateResponse <- wdlService.validateFile(FileContent(request.projectFile.content))
+                  uploadResponse <- wdlService.uploadFile(request.project, request.projectFile, request.version)
+                } yield {
+                  (validateResponse, uploadResponse) match {
+                    case (Right(_), Right(response)) => StatusCodes.OK.intValue -> response
+                    case (Left(_), Right(response))  => StatusCodes.Created.intValue -> response
+                    case (_, Left(response))         => StatusCodes.UnprocessableEntity.intValue -> response.message
+                  }
+                }) {
+                  case Success((status, message)) => complete(status, message)
+                  case Failure(e)                 => complete(StatusCodes.InternalServerError, e.getMessage)
+                }
+            }
+          },
+          get {
+            parameter('projectId.as[String], 'path.as[String], 'version.as[String]) {
+              (projectId, path, version) =>
+                onComplete(projectService.getProjectById(ProjectId(projectId)).flatMap {
+                  case Some(project) => wdlService.getFile(project, Paths.get(path), Some(PipelineVersion(version)))
+                  case None =>
+                    Future.successful(Left(VersioningException(s"Project with ID $projectId does not exist")))
+                }) {
+                  case Success(Left(e)) => complete(StatusCodes.NotFound, e.getMessage)
+                  case Success(_)       => complete(StatusCodes.OK)
+                  case Failure(e)       => complete(StatusCodes.InternalServerError, e.getMessage)
                 }
-              }) {
-                case Success((status, message)) => complete(status, message)
-                case Failure(e)                 => complete(StatusCodes.InternalServerError, e.getMessage)
-              }
+            }
+          },
+          delete {
+            parameter('projectId.as[String], 'path.as[String], 'branchName.as[String], 'commitMessage.as[String]) {
```

Comment:
Why're we specifying such things as `branchName` and `commitMessage` ?
User doesn't know that we have git under the hood, so we should not disclosure internals for anyone.